tools/codegen/src/main.rs

```diff
@@ -161,28 +163,44 @@ fn main() -> Result<()> {
         }
     }
 
-    // Populate license_markdown
+    // Populate license_markdown from the base manifest if present.
     if let Some(license_markdown) = base_info.license_markdown {
         if license_markdown.is_empty() {
             panic!("license_markdown can not be an empty value");
         }
         manifests.license_markdown = license_markdown;
-    } else if let Some(detail) = crates_io_version_detail {
-        if let Some(license) = detail.license {
-            eprintln!("Trying to using license '{license}' from crates.io ...");
-            if let Some(license_markdown) =
-                get_license_markdown(&license, repo, &repo_info.default_branch)
-            {
-                manifests.license_markdown = license_markdown;
+    }
+
+    // Check if the license_markdown is valid.
+    if !manifests.license_markdown.is_empty() {
+        for url in get_license_markdown_urls(&manifests.license_markdown) {
+            if let Err(err) = github_head(&url) {
+                eprintln!("Failed to fetch pre-existing license_markdown {url}: {err}");
+                manifests.license_markdown = String::new();
+                break;
             }
         }
-    } else if let Some(license) = repo_info.license {
-        if let Some(license) = license.spdx_id {
-            eprintln!("Trying to using license '{license}' from github.com ...");
-            if let Some(license_markdown) =
-                get_license_markdown(&license, repo, &repo_info.default_branch)
-            {
-                manifests.license_markdown = license_markdown;
+    }
+
+    // Try to detect license_markdown from crates.io or GitHub.
+    if manifests.license_markdown.is_empty() {
+        if let Some(detail) = crates_io_version_detail {
+            if let Some(license) = detail.license {
+                eprintln!("Trying to verify license '{license}' obtained from crates.io ...");
+                if let Some(license_markdown) =
+                    get_license_markdown(&license, repo, &repo_info.default_branch)
+                {
+                    manifests.license_markdown = license_markdown;
+                }
+            }
+        } else if let Some(license) = repo_info.license {
+            if let Some(license) = license.spdx_id {
+                eprintln!("Trying to verify license '{license}' obtained from github.com ...");
+                if let Some(license_markdown) =
+                    get_license_markdown(&license, repo, &repo_info.default_branch)
+                {
+                    manifests.license_markdown = license_markdown;
+                }
```

Comment:
Isn't the two branches doing the same thing, despite having different logging?

Reply:
yes, and this logic hasnt changed in this PR; the lines have been indented, and the stderr output message has been slightly improved to be more readable.

Reply:
Overhauled this to do the destructuring more neatly and thus DRY up that inner bit of code.